contracts/tfi-factory/src/contract.rs

```diff
@@ -34,11 +34,19 @@ pub fn instantiate(
         return Err(ContractError::InvalidCommission(msg.default_commission));
     }
 
+    let contract_info_query = QueryRequest::Wasm(WasmQuery::ContractInfo {
+        contract_addr: env.contract.address.to_string(),
+    });
+    let contract_info = deps
+        .querier
+        .query::<ContractInfoResponse>(&contract_info_query)?;
+
     let config = Config {
         owner: info.sender,
         token_code_id: msg.token_code_id,
         pair_code_id: msg.pair_code_id,
         default_commission: msg.default_commission,
+        migrate_admin: contract_info.admin,
```

Comment:
What's the difference between `message_info.sender` and `contract_info.admin`? Havent used the former and I'm surprised.